src/lib/krb5/keytab/ktfns.c

```diff
@@ -129,6 +131,51 @@ krb5_kt_have_content(krb5_context context, krb5_keytab keytab)
     return KRB5_KT_NOTFOUND;
 }
 
+static krb5_error_code
+match_entries(krb5_context context, krb5_keytab keytab,
+              krb5_const_principal mprinc)
+{
+    krb5_error_code ret;
+    krb5_keytab_entry ent;
+    krb5_kt_cursor cursor;
+    krb5_boolean match;
+
+    /* Scan the keytab for host-based entries matching accprinc. */
+    ret = krb5_kt_start_seq_get(context, keytab, &cursor);
+    if (ret)
+        return ret;
+    while ((ret = krb5_kt_next_entry(context, keytab, &ent, &cursor)) == 0) {
```

Comment:
`ret` is not read after this assignment - this causes all iteration errors to become `KRB5_KT_NOTFOUND`.

Reply:
Fixed.  Since a double-trinary expression doesn't read well to me, I added an `if (ret) return ret;`.

Reply:
On self-review I realized we could do:

    return (ret == KRB5_KT_END) ? KRB5_KT_NOTFOUND : ret;

with no reference to `match`, because ret will be 0 if we broke out of the loop with a match and KRB5_KT_END if we finished iteration or encountered an iteration error.  However, the current three-line approach may be more obviously correct and therefore worth the extra verbiage.